pkg/controller/persistentvolume/persistentvolume_recycler_controller.go

```diff
@@ -123,10 +205,32 @@ func (recycler *PersistentVolumeRecycler) reclaimVolume(pv *api.PersistentVolume
 			glog.Errorf(errMsg)
 			return fmt.Errorf(errMsg)
 		}
+	} else {
+		glog.V(3).Infof("PersistentVolume[%s] phase %s - skipping: %v", pv.Name, pv.Status.Phase, err)
 	}
 	return nil
 }
 
+func (recycler *PersistentVolumeRecycler) deleteVolume(pv *api.PersistentVolume) error {
+	recycler.removeFailedVolume(pv)
+	return nil
+}
+
+func (recycler *PersistentVolumeRecycler) updateFailedVolume(pv *api.PersistentVolume) {
+	status, found := recycler.failedVolumes[pv.Name]
+	if !found {
+		// First failure, set retryCount to 0 (will be inceremented few lines below)
+		status = failedVolumeStatus{}
+	}
+	status.retryCount += 1
+	status.lastAttempt = time.Now()
+	recycler.failedVolumes[pv.Name] = status
+}
+
+func (recycler *PersistentVolumeRecycler) removeFailedVolume(pv *api.PersistentVolume) {
+	delete(recycler.failedVolumes, pv.Name)
```

Comment:
Do we need `removeFailedVolume`?  This is a one-liner and so is the `deleteVolume` func.  Maybe just `deleteVolume`?